include/envoy/event/dispatcher.h

```diff
@@ -120,11 +120,13 @@ class Dispatcher {
    * empty, it will not use any specific resolvers, but use defaults (/etc/resolv.conf)
    * @param use_tcp_for_dns_lookups if set to true, tcp will be used to perform dns lookups.
    * Otherwise, udp is used.
+   * @param use_apple_api_for_dns_lookups if set to true, Apple APIs will be used to perform dns
```

Comment:
In addition to compiling selectively (which is necessary due to header absence, obviously), I also added this new parameter in line with the tcp parameter's precedence.

The decision I am more ambivalent about is whether the parameter should come from runtime (as the PR has now), or should be added to actual configuration like the tcp config? wdyt? While spanning many files, the change is relatively easy so I am cool with changing if wanted.

Reply:
I don't think we should have this param on the public interface. I would make the runtime switch be internal to the impl/

Reply:
@mattklein123 Would you rather have the runtime switch in the dispatcher function or even deeper in the resolver (by having a wrapper that can create the c ares or the apple resolver). 

I had originally done the former, but ran into a dependency cycle between the dispatcher and runtime. I can look into how challenging ironing that out is.

Reply:
I would just have it inside the creation function to pick which impl to return. You can access runtime via static singleton.

Reply:
> You can access runtime via static singleton

of course! 🤦 update coming in a sec.